packages/app/src/components/Normal/FloatingActionButton/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ReactNode, useCallback, useState } from "react"
+import { FC } from "react"
+import { enhance } from "src/utilities/essentials"
+import { FloatingActionButtonFallback } from "./fallback"
+import { x } from "@xstyled/styled-components"
+import { Minus } from "@styled-icons/boxicons-regular/Minus"
+import { Add } from "@styled-icons/material-sharp/Add"
+import { V } from "src/styles/styleFragments"
+
+export type Menu = {
+  icon: ReactNode
+  onClick: () => void
+}
+
+export type FloatingActionButtonImpureProps = {
+  menus: Menu[]
+}
+
+export const FloatingActionButtonImpure: FC<FloatingActionButtonImpureProps> =
+  enhance<FloatingActionButtonImpureProps>(({ menus }) => {
+    const [isActive, setIsActive] = useState(false)
+
+    const onClickFloatingButton: React.MouseEventHandler<HTMLButtonElement> =
+      useCallback(() => {
+        setIsActive(!isActive)
+      }, [isActive])
+
+    return (
+      <x.div>
+        <FloatingActionButtonPure
+          icon={isActive ? <Minus /> : <Add />}
+          onClick={onClickFloatingButton}
+          active={true}
+          nth={-1}
+        />
+        {menus.map((menu, idx) => (
+          <FloatingActionButtonPure
+            key={idx}
+            icon={menu.icon}
+            onClick={menu.onClick}
+            active={isActive}
+            nth={idx}
```

Comment:
```suggestion
            index={idx}
```
index could be a more appropriate wording

Reply:
nth => index